packages/@aws-cdk/aws-codedeploy/lib/deployment-group.ts

```diff
@@ -108,22 +131,26 @@ export class ServerDeploymentGroup extends ServerDeploymentGroupRef {
     public readonly deploymentGroupName: string;
 
     constructor(parent: cdk.Construct, id: string, props?: ServerDeploymentGroupProps) {
-        super(parent, id);
+        super(parent, id, props && props.deploymentConfig);
+
+        props = props || {};
 
-        this.application = (props && props.application) || new ServerApplication(this, 'Application');
+        this.application = props.application || new ServerApplication(this, 'Application');
 
-        this.role = (props && props.role) || new iam.Role(this, 'Role', {
+        this.role = props.role || new iam.Role(this, 'Role', {
             assumedBy: new cdk.ServicePrincipal('codedeploy.amazonaws.com'),
             managedPolicyArns: ['arn:aws:iam::aws:policy/service-role/AWSCodeDeployRole'],
         });
 
         const resource = new cloudformation.DeploymentGroupResource(this, 'Resource', {
             applicationName: this.application.applicationName,
-            deploymentGroupName: props && props.deploymentGroupName,
+            deploymentGroupName: props.deploymentGroupName,
             serviceRoleArn: this.role.roleArn,
+            deploymentConfigName: props.deploymentConfig &&
+                props.deploymentConfig.deploymentConfigName,
         });
 
-        this.deploymentGroupName = resource.ref;
+        this.deploymentGroupName = resource.ref.toString();
```

Comment:
Don't you `ref`, there should be a strong-name such as `resource.deploymentGroupName`.
Also, FYI, `ref` is already a string.

Reply:
Lol @ "Don't you `ref`" 🙃

Changed to use `deploymentGroupName` :)